tests/tacacs/test_ro_disk.py

```diff
@@ -95,6 +94,16 @@ def test_ro_disk(localhost, duthosts, enum_rand_one_per_hwsku_hostname, creds_al
         
         assert ret, "Failed to ssh as rw user"
 
+        # Force a log rotate to get logs so far recorded.
+        # The read-only state and follow up reboot could wipe off current logs
+        #
+        logging.info("test_ro_disk: Flushing & log rotating to save logs so far")
+        duthost.shell("logger -p local0.notice 'test_ro_disk: Initiating log rotate'")
+        duthost.shell("pkill -HUP rsyslogd")    # To flush logs from daemon to file
+        time.sleep(30)                          # Give a pause before rotate
+        duthost.shell("/usr/sbin/logrotate -f /etc/logrotate.conf > /dev/null 2>&1")
+        time.sleep(30)                          # Pause before making disk RO
+
```

Comment:
I don't get your comments.  
Do you mean, "_test should not add its own log line_" ?
BTW, manual test run succeeded. The Loganalyzer was just looking for its marker only.

Reply:
The log analyzer *first* do the log rotation and *then* put the start-marker. W/o this code change (line #104) of explicit log rotation, all logs since the log rotate by analyzer will be lost, which would include the start-marker set by analyzer.
Logs will be available only upon reboot, which will have end-marker.

So w/o this change, you need to speculate the start of the test, from the end-marker.

Reply:
In short: 
W/o the above code to flush & force log rotation, the test will fail with the following during teardown in loaganalyzer.

```
parallel_run(analyze_logs, [analyzers, markers], {}, duthosts, timeout=120)
...
p_exception = run module extract_log failed, Ansible Results =>
{
    "changed": false, 
   ...21:52:57 was not found in /var/log',), <traceback object at 0x7ff388846b90>)"
}
p_exitcode = 1
p_traceback = 'Traceback (most recent call last):\n  File "/var/src/sonic-mgmt-int/tests/common/helpers/parallel.py", line 28, in ru...o_disk[str-dx010-acs-4].2021-10-09-21:52:57 was not found in /var/log\',), <traceback object at 0x7ff388846b90>)"\n}\n'
process    = {'exception': (run module extract_log failed, Ansible Results =>
{
    "changed": false, 
   ...21:52:57 was not found...cs-4].2021-10-09-21:52:57 was not found in /var/log\',), <traceback object at 0x7ff388846b90>)"\n}\n'), 'exit_code': 1}
process_name = 'analyze_logs--<MultiAsicSonicHost> str-dx010-acs-4'

```
'Sorry if I have not explained in detail.
 ```
Timepoint 0: Test starts
 
                1: Log analyzer does rotation; This creates a new log file for future logs
                            Log rotate happens before writing start marker

                  2: Loganalyzer writes "start: <test name>" as start marker
                            This would be in memory, **not flushed into file yet** 
                            Reliable flushing needs write followed by file-handle close
                            The rsyslog will not do this until logrotate or SIGHUP

                  3. The test code makes disk as read-only
                 
                         From here on no disk write can happen.
                         This means the logs written since step 1, which includes start marker will not get written into file
              
                  4. Reboot device
                      The logging starts

                 5. Loganalyzer writes end marker
                     You can see this in logs,
```